redis/commands/search/commands.py

```diff
@@ -857,3 +857,245 @@ def syndump(self):
         """  # noqa
         raw = self.execute_command(SYNDUMP_CMD, self.index_name)
         return {raw[i]: raw[i + 1] for i in range(0, len(raw), 2)}
+
+
+class AsyncSearchCommands(SearchCommands):
```

Comment:
I could see this becoming too difficult to maintain. WDYT about moving this into async_commands.py?

Reply:
That might get messy once you start thinking about the other Redis modules like AI and their own async counterparts.

It might help to choose either of these options instead:
1. Under the current commands folder, create an asyncio module
2. Under the asyncio module, create a new commands directory

Reply:
It's true - commands are becoming unwieldy. Is there a more accepted pattern for maintaining async support?  What about the following pattern (note my being new to async), yes this might break things:

1. Create functions for each command (i.e set) prefixed with double underscores to denote private
2. Create their implementations in both async and sync versions, with execute_command being different, to support async/await.

Gut check: Does this seem painful as you'd now need to create one implementation and two wrappers per function? Alternatively we could generate #2 quite frankly, easily enough.

@Andrew-Chen-Wang thoughts? I'm just spitballing here.

Reply:
I think the only patterns that I've seen adopted are to use an a-  or async- prefix (eg Python itself, django) or to have separated mixins (eg Elasticsearch-py).

And mm the current implementation of an abstraction of execute_command, to me, is the most viable method of maintainence. I think adding too many ways of separation between sync and async will make redis-py more bloated and unmaintainable.

Reply:
Agreed. I think that's out of scope for this (obviously!) but let's not lose track. I'm literally going to open a GH issue, and link to this comment.